File: easy_2/beer-song/ruby/v2/beer_song.rb
class BeerSong
  PHRASE="%d bottles of beer on the wall, %d bottles of beer.\n"
  PHRASE1="Take one down and pass it around, %d bottles of beer on the wall.\n"

  def initialize; end

  def verse(num)
    case
    when num > 2
      phrase = PHRASE % [num, num]
      phrase1 = PHRASE1 % (num - 1)
    when num == 2
      phrase = PHRASE % [num, num]
      phrase1 = "Take one down and pass it around, 1 bottle of beer on the wall.\n"
    when num == 1
      phrase = "1 bottle of beer on the wall, 1 bottle of beer.\n"
      phrase1 = "Take it down and pass it around, no more bottles of beer on the wall.\n"
    when num == 0
      phrase = "No more bottles of beer on the wall, no more bottles of beer.\n"
      phrase1 = "Go to the store and buy some more, 99 bottles of beer on the wall.\n"
    else
      return "Hey! You can't play this game with that number!!"
    end
    phrase + phrase1
  end

  def verses(first, last)
    beer_song = verse(first)
    next_verse = first - 1
    next_verse.downto(last) do |num|
      beer_song.concat("\n" + verse(num))
    end
    beer_song
  end

  def lyrics
    verses(99, 0)
  end
end

File: easy_2/beer-song/js/beer_song.js
export default class BeerSong {
  constructor() {
    this.phraseOne = "$ bottles of beer on the wall, $ bottles of beer.\n";
    this.phraseTwo = "Take one down and pass it around, $ bottles of beer on the wall.\n";
  }
  verse(num) {
    if (num < 0) { return 'You so baaadd!! That aint done with this song.' }
    let songVerse;
    switch(num) {
      case 0:
        songVerse = "No more bottles of beer on the wall, no more bottles of beer.\n" +
                    "Go to the store and buy some more, 99 bottles of beer on the wall.\n";
        break;
      case 1:
        songVerse = "1 bottle of beer on the wall, 1 bottle of beer.\n" +
                    "Take it down and pass it around, no more bottles of beer on the wall.\n";
        break;
      case 2:
        songVerse = "2 bottles of beer on the wall, 2 bottles of beer.\n" +
                    "Take one down and pass it around, 1 bottle of beer on the wall.\n";
        break;
      default:
        songVerse = this.phraseOne.replace(/[$]/g, num);
        songVerse = songVerse.concat(this.phraseTwo.replace("$", num - 1))
    }
    return songVerse;
  }

  verses(firstVerse, lastVerse) {
    let excerpt = this.verse(firstVerse);

    for (let nextVerse = firstVerse - 1; nextVerse >= lastVerse; nextVerse -= 1) {
      excerpt += ("\n" + this.verse(nextVerse));
    }
    return excerpt;
  }

  lyrics() {
    return this.verses(99, 0);
  }
}
